test/test_operators.jl

```diff
@@ -2,323 +2,394 @@ using Test
     end
   end
-end
 
-@testset "test shift operators in 2D on Tree" begin
-  g = named_comb_tree((3, 3))
-  s = continuous_siteinds(g; map_dimension=2)
-  L = length(dimension_inds(s, 2))
-  delta = 2.0^(-1.0 * L)
-  x = 0.5
-  ys = [0.0, delta, 0.25, 0.5, 0.625, 0.875, 1.0 - delta]
-  ψ_fx = poly_itn(s, [1.0, 0.5, 0.25]; dimension=1)
-  ψ_fy = cos_itn(s; dimension=2)
-  ψ_fxy = ψ_fx + ψ_fx
-
-  forward_shift_dirichlet = forward_shift_op(
-    s; boundary="Dirichlet", dimension=2, truncate_kwargs=(; cutoff=1e-10)
-  )
-  backward_shift_dirichlet = backward_shift_op(
-    s; boundary="Dirichlet", dimension=2, truncate_kwargs=(; cutoff=1e-10)
-  )
-  forward_shift_pbc = forward_shift_op(
-    s; boundary="Periodic", dimension=2, truncate_kwargs=(; cutoff=1e-10)
-  )
-  backward_shift_pbc = backward_shift_op(
-    s; boundary="Periodic", dimension=2, truncate_kwargs=(; cutoff=1e-10)
-  )
-  forward_shift_neumann = forward_shift_op(
-    s; boundary="Neumann", dimension=2, truncate_kwargs=(; cutoff=1e-10)
-  )
-  backward_shift_neumann = backward_shift_op(
-    s; boundary="Neumann", dimension=2, truncate_kwargs=(; cutoff=1e-10)
-  )
-
-  ψ_fxy_pshift_dirichlet = operate(forward_shift_dirichlet, ψ_fxy; cutoff=1e-12)
-  ψ_fxy_mshift_dirichlet = operate(backward_shift_dirichlet, ψ_fxy; cutoff=1e-12)
-  ψ_fxy_pshift_pbc = operate(forward_shift_pbc, ψ_fxy; cutoff=1e-12)
-  ψ_fxy_mshift_pbc = operate(backward_shift_pbc, ψ_fxy; cutoff=1e-12)
-  ψ_fxy_pshift_neumann = operate(forward_shift_neumann, ψ_fxy; cutoff=1e-12)
-  ψ_fxy_mshift_neumann = operate(backward_shift_neumann, ψ_fxy; cutoff=1e-12)
-
-  for y in ys
-    if y + delta < 1
-      fxy_xyplus = calculate_fxyz(ψ_fxy, [x, y + delta])
-      @test fxy_xyplus ≈ calculate_fxyz(ψ_fxy_pshift_dirichlet, [x, y]) atol = 1e-8
-      @test fxy_xyplus ≈ calculate_fxyz(ψ_fxy_pshift_pbc, [x, y]) atol = 1e-8
-      @test fxy_xyplus ≈ calculate_fxyz(ψ_fxy_pshift_neumann, [x, y]) atol = 1e-8
-    elseif y == 1.0 - delta
-      @test calculate_fxyz(ψ_fxy_pshift_dirichlet, [x, y]) ≈ 0.0 atol = 1e-8
-      @test calculate_fxyz(ψ_fxy_pshift_pbc, [x, y]) ≈ calculate_fxyz(ψ_fxy, [x, 0.0]) atol =
-        1e-8
-      @test calculate_fxyz(ψ_fxy_pshift_neumann, [x, y]) ≈
-        calculate_fxyz(ψ_fxy, [x, 1.0 - delta]) atol = 1e-8
+  @testset "test boundary operator in 2D on Tree" begin
+    g = named_comb_tree((3, 3))
+    s = continuous_siteinds(g; map_dimension=2)
+    L = length(dimension_inds(s, 2))
+    delta = 2.0^(-1.0 * L)
+    lastDigit = 1 - delta
+
+    ys = [0.0, delta, 0.25, 0.5, 0.625, 0.875, 1.0 - delta]
+    ψ_fx = poly_itn(s, [1.0, 0.5, 0.25]; dimension=1)
+    ψ_fy = cos_itn(s; dimension=2)
+    ψ_fxy = ψ_fx + ψ_fx
+
+    Zo = map_to_zero_operator(s, [0, lastDigit, 0, lastDigit], [1, 1, 2, 2])
+    @testset "corner boundary test" begin
+      for p1 in [0, lastDigit]
+        for p2 in [0, lastDigit]
+          p = ttn(itensornetwork(delta_xyz(s, [p1, p2])))
+          @test inner(p', Zo, p) ≈ 0.0
```

Comment:
Should be able to strip the `ttn` wrapper and just do
```
p = itensornetwork(delta_xyz(s, [p1, p2]))
@test inner(p, Zo, p) ≈ 0.0
```
or is this a case where BP is giving 0 so it bugs out? If so keep as is but maybe add a TODO that we change it back once the BP thing is fixed.

Reply:
This is indeed giving the BP bug of NaNs. Will change to exact and leave a TODO